### ISSUE TITLE
[0.16] Comments at the end of a pattern matching block raising a `syntax problem`

### ISSUE THREAD

**Issue by Warry** [elm-lang/elm-compiler#1099]
(With 0.16-alpha2)
Ending a pattern matching block with a comment fails:

```elm
type Foo = Bar | Baz -- | Buzz

fooToString : Foo -> String
fooToString x =
  case x of
    Bar -> "bar"
    Baz -> "baz"
    -- Buzz -> "buzz"
```

```
-- SYNTAX PROBLEM ------------------------------------------------------
I ran into something unexpected when parsing your code!
I am looking for one of the following things:
    whitespace
```

This was working fine on the previous alpha.

**Comment by evancz** [comment]
Oh, nice find! This is because of 6e1a9e1694ad150a8ac37f925d2a3009aaefa3af fixing #1083. I'll take another look at it for a fix.

**Comment by evancz** [comment]
Interesting, it seems that the amount of whitespace after the comment changes things. If the end of file comes at the end of the line comment, this is the error:

```
-- SYNTAX PROBLEM ------------------------------------------------------ tmp.elm

I ran into something unexpected when parsing your code!

8│     -- Buzz -> "buzz"
                        ^
I am looking for one of the following things:

    "\n"
    "\r\n"
    end of input
    whitespace
```

**Comment by evancz** [comment]
Alright, this should be fixed by #1101, thanks for the report!
